### DISCUSSION TITLE
General Disaggregators Feedback

### DISCUSSION THREAD

**Discussion by NimaBoscarino** [huggingface/disaggregators#23] [General]
Hello! Please feel free to comment in this thread if you have ideas for new disaggregation modules, suggestions for how to improve the existing ones, general API feedback, or anything else 🤗

**Comment by davidberenstein1957** [comment]
Hi @NimaBoscarino, I love the idea behind the library, but I would love to see the magic happening in the kind of black box. 

Some datasets and language models are required and poof, we`ve got the labels. 
To me, it makes sense to also see the matches that cause these labels to be assigned or at least have the option to do so. Potentially even approach it as a sort of token-classification approach that can be visualized in a similar manner as https://explosion.ai/blog/spancat. 

I have worked on a similar approach/reasoning for an analysis for the Dutch dispatch centre, which led me to create https://github.com/Pandora-Intelligence/concise-concepts. If you need help, I would love to contribute and help maintain/improve the package.

**Reply by NimaBoscarino** [reply]
Agreed, I can totally see how the current "black box" design isn't ideal! It's really important to me to make all the disaggregations as transparent and as customizable as possible, and I would _love_ your help with the package 🤗 I see that you've sent me a DM, so I'll message you and we'll set something up!
